starter-code/src/main.js

```diff
@@ -40,8 +120,64 @@ $(document).ready(function(){
 
   // Bind the click event of each element to a function
   $('.back').click(function () {
-    // TODO: write some code here
+    console.log(clicked); // for my revision
+
+    $(this).hide(); // hide div wit back function 
+    $(this).next().toggleClass("back"); // but also give .back class to div next to .back. So it has now background image + width and height
+    $(this).toggleClass("blocked"); // once it is revealed it can not be turned on back just like that...keep it open until next card is revealed.
+    clicked.push($(this)); //put "this" (complete HTML tag) in clicked array.
+    console.log(clicked);
+    picked.push($(this).attr("name")); // put in picked array only valeu of name attr.
+    console.log(picked);
+    var checkedPicked = checkPickedLength(picked); // global function which checks if picked array has two elements. In that case compare it.
+    //if it is true go tu if statement defined below.
+
+    if (checkedPicked == true) {
+
+      $(".back").toggleClass("blocked"); // add class="blocked"  to all elements without it. 
+      var checkedEqual = memoryGame.checkIfPair(picked[0], picked[1]); // moment to check if to strings from picked array are the same.
+      //we call method defined in memoryGame prototype.
+      console.log(checkedEqual);
+      console.log(memoryGame.pairsClicked);
+      console.log(memoryGame.pairsGuessed);
+      $("#pairs_clicked").text(memoryGame.pairsClicked); //"checkIfPair" method changes memoryGame properties, every time two cards are clicked. 
+      //Every time properties are changed we should see it on page.
+      $("#pairs_guessed").text(memoryGame.pairsGuessed); //"checkIfPair" method changes memoryGame properties, every time two cards are clicked. 
+      //Every time properties are changed we should see it on page.
+
+
+      if (checkedEqual == false) { //if checkIfEqual returns false flip cards to back after 1 sec.
+        setTimeout(function () {
+          $(clicked[0].next()).toggleClass("back"); // it will remove .back from .front
```

Comment:
To toggle back the cards  you've used several lines which can be done a little bit more eficient with the following lines. In order to have this working you should select the whole html element that you pushed to the clicked array on line 129:

this.pickedCards[0].children().toggleClass('front back');
this.pickedCards[1].children().toggleClass('front back');